nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -153,6 +163,24 @@ let
           example = "/var/lib/hello_world";
         };
 
+        containerDependencies = mkOption {
```

Comment:
I'm not sure  this option is a good idea since it is fragile: you are only checking the container is started but you are not sure the container application is ready. If you depend on a container, this is because you want to consume an application of this container and it would be better to be sure this application is ready. To achieve this, you would have to implement a probe to check your service is **really** responding.

Also, if a user really wants to do that, it is still possible for him to add such kind of dependencing by manipulating the associated systemd service directly.

Reply:
I think you might be reading a little too much into this. I have no intentions of providing support for service health checks here. This just achieves some basic unit ordering and takes down dependent units when one fails.

Reply:
So, ok. Let's do it.
One last thing here: what about naming this option `dependsOn` since this is the nomenclature used in docker-compose file?

Reply:
Fair enough, I've renamed it.